# Author: Damyan Ivanov <dmn@debian.org>
# Description: disable the tcp listening service that is to be used to controll
#              running instance of padre. It lacks any sort of authentication
#              and binds t 127.0.0.1
#              The proper way to fix that would be to make it listen on an Unix
#              domain socket somewhere in $HOME ensuring the right permissions,
#              but I found no evidence if Wx::SocketServer supporting UNIX
#              domain sockets.
# Upstream-Bug: http://padre.perlide.org/ticket/313
--- a/lib/Padre/Wx/Main.pm
+++ b/lib/Padre/Wx/Main.pm
@@ -1068,6 +1068,10 @@ sub single_instance_start {
 	# check if server is already started
 	return 1 if $self->single_instance_running;
 
+    warn "single instance server is disabled in the Debian package\n";
+    warn "see /usr/share/doc/padre/README.Debian for rationale.\n";
+    return 0;
+
 	# Create the server
 	require Wx::Socket;
 	$self->{single_instance} = Wx::SocketServer->new(
